airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -239,12 +240,34 @@ public AirbyteMessageConsumer getConsumer(JsonNode config,
       }
       final WriteDisposition syncMode = getWriteDisposition(configStream.getDestinationSyncMode());
 
-      writeConfigs.put(AirbyteStreamNameNamespacePair.fromAirbyteSteam(stream),
-          new BigQueryWriteConfig(TableId.of(schemaName, tableName), TableId.of(schemaName, tmpTableName), writer, syncMode, schema));
+      if (isGcsUploadingMode) {
+        GcsDestinationConfig gcsDestinationConfig = GcsDestinationConfig
+            .getGcsDestinationConfig(BigQueryUtils.getGcsJsonNodeConfig(config));
+        GcsCsvWriter gcsCsvWriter = initGcsWriter(gcsDestinationConfig, configStream);
+        gcsCsvWriter.initialize();
+
+        writeConfigs.put(AirbyteStreamNameNamespacePair.fromAirbyteSteam(stream),
+            new BigQueryWriteConfig(TableId.of(schemaName, tableName), TableId.of(schemaName, tmpTableName),
+                writer, syncMode, schema, gcsCsvWriter, gcsDestinationConfig));
+      } else {
+        writeConfigs.put(AirbyteStreamNameNamespacePair.fromAirbyteSteam(stream),
+            new BigQueryWriteConfig(TableId.of(schemaName, tableName), TableId.of(schemaName, tmpTableName),
+                writer, syncMode, schema, null, null));
+      }
+
     }
     // write to tmp tables
     // if success copy delete main table if exists. rename tmp tables to real tables.
-    return getRecordConsumer(bigquery, writeConfigs, catalog, outputRecordCollector);
+    return getRecordConsumer(bigquery, writeConfigs, catalog, outputRecordCollector, isGcsUploadingMode, isKeepFilesInGcs);
+  }
+
+  private GcsCsvWriter initGcsWriter(GcsDestinationConfig gcsDestinationConfig,
+                                     ConfiguredAirbyteStream configuredStream)
+      throws IOException {
+    Timestamp uploadTimestamp = new Timestamp(System.currentTimeMillis());
+
+    AmazonS3 s3Client = GcsS3Helper.getGcsS3Client(gcsDestinationConfig);
```

Comment:
Why are we creating AmazonS3 client if we are using Google cloud storage for loading data

Reply:
GCS is compatible with the Amazon S3 client. By reusing the S3 client, we can reuse the related code.

Reply:
Hi @subodh1810.  I just partially re-used our already existing destination-gcs module that had been created before I started working on this ticket.  I had some conversation with @tuliren  and he confirmed that S3 client was also used for destination-gcs as it mostly works with both except for some minor cases.